### PR TITLE
Laravel 10.x Compatibility

### PR DESCRIPTION
**Pull request by laravel-shift**
This is an automated pull request from [Shift](https://laravelshift.com) to update your package code and dependencies to be compatible with Laravel 9.x.

**Before merging**, you need to:

- Checkout the `l10-compatibility` branch
- Review **all** comments for additional changes
- Thoroughly test your package

If you do find an issue, please report it by commenting on this PR to help improve future automation.

### CODE REVIEW

**Comment by laravel-shift** [comment]
:warning: Shift detected GitHub Actions which run jobs using a version `matrix`. Shift attempted to [update your configuration](https://github.com/beyondcode/laravel-mailbox/pull/113/commits/9cbb8e511f6c632c9f66f1ef510a6fa8a55ae5f6?w=1) for Laravel 10. However, you should review these changes to ensure the desired combination of versions are built for your package.

**Comment by laravel-shift** [comment]
:alembic: **Using this package?** If you would like to help test these changes or believe them to be compatible, you may update your project to reference this branch.

To do so, temporarily add Shift's fork to the `repositories` property of your `composer.json`:

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/laravel-shift/laravel-mailbox.git"
        }
    ]
}
```

Then update your dependency constraint to reference this branch:

```json
{
    "require": {
        "beyondcode/laravel-mailbox": "dev-l10-compatibility",
    }
}
```

Finally, run: `composer update`

**Comment by jonnott** [comment]
@mpociot This PR looks like it could contain all that's needed to support Laravel 10.x, based on a review of the changes that were made to support L9.x.

Could this be merged soon and a new version tagged?

**Comment by siebsie23** [comment]
> @mpociot This PR looks like it could contain all that's needed to support Laravel 10.x, based on a review of the changes that were made to support L9.x.
> 
> Could this be merged soon and a new version tagged?

My fork adds support for Laravel 10 while we are waiting for the official package to update.

File: composer.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
         "illuminate/routing": "^6.0|^7.0|^8.0|^9.0|^10.0",
         "illuminate/support": "^6.0|^7.0|^8.0|^9.0|^10.0",
         "willdurand/email-reply-parser": "^2.8",
-        "zbateson/mail-mime-parser": "^1.1|^2.3"
```
Comment (jonnott) [inline composer.json]: @mpociot What's the conflict/issue with mail-mine-parser 2.x ?

Comment (mpociot) [reply composer.json]: I just wanted to handle this in a separate PR 👍

**Pull request merged**